### ISSUE TITLE
Add methods to extract translation and rotation from `Mat4` 

### ISSUE THREAD

**Issue by thenlevy** [termhn/ultraviolet#79]
Given a `Mat4` of the form `M = T * R` where `T` is a translation matrix and `R` is a rotation matrix, I would like to be able to retrieve `T` as a `Vec3` and `R` as a `Rotor3` from `M`. 

Doing this would also make it possible to implement a `Mat4 -> Isometry3` conversion.

Retrieving `T` is easy it is the tranlsation given by `M.transform_point3(Vec3::zero())` 
To get `R` one can do 
```rust 
Mat3::new(
  M.transform_vec3(Vec3::unit_x()),
  M.transform_vec3(Vec3::unit_y()),
  M.transform_vec3(Vec3::unit_z()),
).into_rotor3()
```
But maybe there is a better way.

What are your opinion about adding these methods ?
One problem that not all `Mat4`s are of the product of a translation and a rotation so this factorization would not always make sense.

**Comment by termhn** [comment]
Sounds like good things to have. As far as it not being always valid,
that's fine as long as it's documented properly. Also you can extract
translation simply by doing `mat[3].truncated()`

**Comment by termhn** [comment]
Closed by #80
